final_task/rss_reader/article.py

```diff
@@ -0,0 +1,68 @@
+from string_operations import *
+import logging
+import datetime
+
+class Article:
+    """single article class"""
+
+    def __init__(self, parsed, source):
+        """receive parsed article and extracts data from it"""
+        self.title = make_string_readable(parsed.title)
+        self.link = parsed.link
+        self.feed_link = source
+        self.published = extract_date(parsed)
+
+        summary_ = extract_topic_info_from_summary(parsed.summary)
+        self.summary = make_string_readable(summary_)
+        self.media = parsed.media_content
+
+        description_ = extract_image_info_from_summary(parsed.summary)
+        self.media_description = make_string_readable(description_)
+
+    def print_readable_article(self):
+        """print article to stdout in human-readable format"""
+        print("_" * 79)
+        date = self.published
+        print('{}/{}/{}, {}:{}:{}\n'.format(
+            date.tm_year,
+            date.tm_mon,
+            date.tm_mday,
+            date.tm_hour,
+            date.tm_min,
+            date.tm_sec
+        ))
+
+        cutted_title = cut_string_to_length_with_space(self.title, 77)
+        for str_number, string in enumerate(cutted_title):
+            if str_number + 1 == len(cutted_title):
+                print(string, [1])
+            else:
+                print(string)
+
+        # images description and their links numbers (like [2] - [5])
+        str_number_of_img = ' '
+        if len(self.media) > 1:
+            str_number_of_img = ' - [{}]'.format(len(self.media) + 1)
+        print('\n\nImages:\n{} [2]{}\n'.format(self.media_description, str_number_of_img))
+
+        cutted_summary = cut_string_to_length_with_space(self.summary, 79)
+        for string in cutted_summary:
+            print(string)
+
+        # Links of article and images
+        print('\n\nLinks:\n[1]', self.link)
+        for number, img in enumerate(self.media):
+            print('[{}]'.format(number+2), img['url'])
```

Comment:
It is better to use `f"..."` instead of `.format` in most cases

Reply:
Fixed